### PR TITLE
Use actual message data word count instead of schema's

### PR DESCRIPTION
**Pull request by jbchouinard**
The code assumes that the message has the number of data words
as defined in the schema.

This breaks backward compatibility because messages with extra
data fields appended are supposed to be parseable by older
schemas.

### CODE REVIEW

**Comment by jbchouinard** [comment]
https://github.com/webarmor/vmcode/pull/279

File: capnp.lua [hunk 1]
```diff
@@ -256,7 +256,7 @@ end
 
 function _M.read_text(buf, header, T, offset, default)
     local res
-    local data_off, size, num = _M.read_listp_struct(buf, header, T, 2)
```
Comment (KenMacD) [inline capnp.lua]: Should we also be passing `dw` in to `read_text()`? If we use the `T` values here then it seems like we might end up with the same bug. (Also fixing the compiler calls to this).

Comment (jbchouinard) [reply capnp.lua]: I also missed some stuff that needs to be changed in handwritten_capnp.lua

Comment (jbchouinard) [reply capnp.lua]: And maybe I can add new tests

Comment (KenMacD) [reply capnp.lua]: I'm not sure what the deal with `handwritten_capnp.lua` is.

Comment (jbchouinard) [reply capnp.lua]: I don't know either.. it's not tested and it's not used anywhere

**Pull request closed without merging**